homeassistant/components/swiss_public_transport/coordinator.py

```diff
@@ -74,8 +77,16 @@ def nth_departure_time(self, i: int) -> datetime | None:
         return None
 
     async def _async_update_data(self) -> list[DataConnection]:
+        return await self.fetch_connections(limit=SENSOR_CONNECTIONS_COUNT)
+
+    async def fetch_connections(self, limit: int) -> list[DataConnection]:
+        """Fetch connections using the opendata api."""
+        self._opendata.limit = limit
         try:
             await self._opendata.async_get_data()
+        except OpendataTransportConnectionError as e:
+            _LOGGER.warning("Connection to transport.opendata.ch cannot be established")
+            raise OpendataTransportConnectionError from e
```

Comment:
Just re-raise makes more sense IMO
```suggestion
        except OpendataTransportConnectionError:
            _LOGGER.warning("Connection to transport.opendata.ch cannot be established")
            raise
```